dev/snippets/lib/configuration.dart

```diff
@@ -67,8 +67,9 @@ class Configuration {
   /// dartdoc.
   Directory get templatesDirectory => Directory(path.join(configDirectory.path, 'templates'));
 
-  /// Gets the skeleton file to use for the given [SnippetType].
-  File getHtmlSkeletonFile(SnippetType type) {
-    return File(path.join(skeletonsDirectory.path, '${getEnumName(type)}.html'));
+  /// Gets the skeleton file to use for the given [SnippetType] and DartPad preference.
+  File getHtmlSkeletonFile(SnippetType type, bool useDartPad) {
```

Comment:
It just occurred to me that you could just make a new `SnippetType` and use it here.

Then you can just add something like this to the top-level dartdoc_options.yaml:
```yaml
    dartpad:
      command: ["dev/snippets/lib/main.dart", "--type=dartpad"]
      description: "Creates sample code documentation output from embedded documentation samples and displays it in DartPad."
```

Instead of adding an argument to the location, you change the tool name from `snippet` to `dartpad`.  It has the advantage of allowing you to change the options for all instances of `dartpad` samples in one place without affecting the `snippet` samples.

Reply:
I did think about doing it that way once I got into the code, but wasn't sure if it made sense. The only real difference between using DartPad or not is just choosing the right skeleton file. A flag seemed like a suitably small way to add that detail, while a new snippet type would have implied changes for the generated code that's stored on the server (I'd need a fall through on the giant switch statement in `generate`, for example, since the new type would still need to result in a saved snippet file).

This is code you're going to have to maintain more than me, though, so if you feel it's definitely better the other way, I'll trust you and refactor it.

Reply:
Actually, I think you can do both, in a way.  Change the `dartdoc_options.yaml` to be this:

```yaml
    dartpad:
      command: ["dev/snippets/lib/main.dart", "--type=application", "--dartpad"]
      description: "Creates sample code documentation output from embedded documentation samples and displays it in DartPad."
```

And leave the flag.  That way, you can still change the options for all the dartpad examples in one place, but don't need special handling in the code for a new type.

Reply:
SGTM. I've updated the code to implement this strategy.